### DISCUSSION TITLE
Pretty visualisation of quantum circuits

### DISCUSSION THREAD

**Discussion by maremnev** [sympy/sympy#27412] [General]
Hello!

I made up visual cirquit constructor in jupyer notebook which looks like qiskit.
<img width="767" alt="Image" src="https://github.com/user-attachments/assets/a3b61665-078a-4b6b-8cf8-adb99d332d41" />

The visualisator able to drag and dpop for the gates.

I want to incapsulate it into sympy.physics.quantum.

Who can help me to said the west way to ember my visualisator into quantum ?

**Comment by oscarbenjamin** [comment]
Maybe @ellisonbg knows whether it makes sense to add this.

**Comment by ellisonbg** [comment]
Thanks @oscarbenjamin for the ping and thanks to @maremnev for opening the issue. It would be great to see the effort move forward and get into the hands of users. I can definitely see why users of `sympy.physics.quantum` would be interested in this. At the same time, the develop experience (build system, testing, knowledge & expertise) for a UI like this is pretty different from the core of sympy and the release cadence would also be quite different. Because of that, I think it is best to build this outside the main sympy repo/package. I can think of two options: 1) @maremnev could create a separate repo in their own org, and publish this as a separate package (non-sympy affiliated); 2) the sympy maintainers could approve this as a project "under incubation" and create a new repo under the sympy org for this. I don't have strong opinions about which of these would be better.

**Comment by oscarbenjamin** [comment]
> 2) the sympy maintainers could approve this as a project "under incubation" and create a new repo under the sympy org for this.

I think that if doing this then it would still be better for it to be created as a standalone project first. If there was a desire to move it into the sympy github organisation then that could be done after when seeing the completed project.

**Comment by ellisonbg** [comment]
Sounds good @oscarbenjamin 

@maremnev let us know if you have any questions.

**Comment by maremnev** [comment]
@ellisonbg and @oscarbenjamin, thanks for answer. I guess it will be nice idea to create separated project and use sympy as a calculation kernel.

**Comment by oscarbenjamin** [comment]
Okay, great. I'm going to move this issue to a discussion since I don't think that there is anything to be changed in sympy.
